src/ImageSharp/Color/Color.cs

```diff
@@ -240,28 +261,38 @@ public TPixel ToPixel<TPixel>()
             Span<TPixel> destination)
             where TPixel : unmanaged, IPixel<TPixel>
         {
-            ReadOnlySpan<Rgba64> rgba64Span = MemoryMarshal.Cast<Color, Rgba64>(source);
-            PixelOperations<TPixel>.Instance.FromRgba64(configuration, rgba64Span, destination);
+            Guard.DestinationShouldNotBeTooShort(source, destination, nameof(destination));
+            for (int i = 0; i < source.Length; i++)
+            {
+                destination[i] = source[i].ToPixel<TPixel>();
+            }
         }
 
         /// <inheritdoc />
         [MethodImpl(InliningOptions.ShortMethod)]
         public bool Equals(Color other)
         {
-            return this.data.PackedValue == other.data.PackedValue;
+            if (this.boxedHighPrecisionPixel is null && other.boxedHighPrecisionPixel is null)
+            {
+                return this.data.PackedValue == other.data.PackedValue;
+            }
+
+            return this.ToVector4().Equals(other.ToVector4());
```

Comment:
This implementation is not correct when the pixel is really a high precision one. I think we should expect custom `IPixel` implementations to override `object.Equals(object)`, like our pixels do. This is a practice `struct` equality implementers should follow IMO.

Reply:
Yeah wasn’t sure about that.